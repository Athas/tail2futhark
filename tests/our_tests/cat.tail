let v0:[int]2 = [[1,2],[3,4],[5,6]] in
let v1:[int]2 = [[7,8],[9,10],[11,12]] in
let v2:[int]2 = cat(v0,v1) in
let v3:[int]0 = reduce(addi,0,(reduce{addi,0,v2})) in
i2d(v3)
