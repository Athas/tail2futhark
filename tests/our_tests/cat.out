fun real main() =
  let t_v1 = reshape((3,2),reshape1_int((3 * (2 * 1)),reshape(((size(0,[1,2,3,4,5,6]) * 1)),[1,2,3,4,5,6]))) in
  let t_v2 = reshape((3,2),reshape1_int((3 * (2 * 1)),reshape(((size(0,[7,8,9,10,11,12]) * 1)),[7,8,9,10,11,12]))) in
  let t_v3 = concat(t_v1,t_v2) in
  let t_v6 = reduce(+,0,[size(0,t_v3),size(1,t_v3)]) in
  toReal(t_v6)